core/iwasm/include/wasm_c_api.h

```diff
@@ -181,24 +181,28 @@ typedef union MemAllocOption {
 struct wasm_config_t {
     mem_alloc_type_t mem_alloc_type;
     MemAllocOption mem_alloc_option;
-    bool linux_perf_support;
+#if WASM_ENABLE_LINUX_PERF != 0
+    bool enable_linux_perf;
+#endif
     /*TODO: wasi args*/
 };
 
 /*
  * by default:
  * - mem_alloc_type is Alloc_With_System_Allocator
  * - mem_alloc_option is all 0
- * - linux_perf_support is false
+ * - enable_linux_perf is false
  */
 WASM_API_EXTERN own wasm_config_t* wasm_config_new(void);
 
 // Embedders may provide custom functions for manipulating configs.
 WASM_API_EXTERN own wasm_config_t*
 wasm_config_set_mem_alloc_opt(wasm_config_t *, mem_alloc_type_t, MemAllocOption *);
 
+#if WASM_ENABLE_LINUX_PERF != 0
```

Comment:
Same as above, had better remove the macro and add comment that this API requires building WAMR with linux perf enabled.